ethereum/statetransition/src/test/java/tech/pegasys/teku/statetransition/datacolumns/retriever/SampleSidecarRetrieverTest.java

```diff
@@ -0,0 +1,169 @@
+
+  List<UInt64> nodeCustodyColumns(UInt256 nodeId) {
+    return miscHelpers.computeCustodyColumnIndexes(
+        nodeId, custodyCountSupplier.getCustodyCountForPeer(nodeId));
+  }
+
+  Stream<UInt256> craftNodeIds() {
+    return IntStream.iterate(0, i -> i + 1).mapToObj(UInt256::valueOf);
+  }
+
+  Stream<UInt256> craftNodeIdsCustodyOf(UInt64 custodyColumn) {
+    return craftNodeIds().filter(nodeId -> nodeCustodyColumns(nodeId).contains(custodyColumn));
+  }
+
+  Stream<UInt256> craftNodeIdsNotCustodyOf(UInt64 custodyColumn) {
+    return craftNodeIds().filter(nodeId -> !nodeCustodyColumns(nodeId).contains(custodyColumn));
+  }
+
+  //  private void advanceTime(Duration delta) {
+  //    stubTimeProvider.advanceTimeBy(delta);
+  //    stubAsyncRunner.executeDueActionsRepeatedly();
+  //  }
+
+  private void advanceTimeGradually(Duration delta) {
+    for (int i = 0; i < delta.toMillis(); i++) {
+      stubTimeProvider.advanceTimeBy(Duration.ofMillis(1));
+      stubAsyncRunner.executeDueActionsRepeatedly();
+    }
+  }
+
+  @Disabled
+  @Test
+  void sanityTest() {
+    UInt64 columnId = UInt64.valueOf(1);
+
+    List<TestPeer> custodyPeers =
+        craftNodeIdsCustodyOf(columnId)
+            .limit(2)
+            .map(nodeId -> new TestPeer(stubAsyncRunner, nodeId, Duration.ofMillis(100)))
+            .peek(testPeer -> testPeer.setCurrentRequestLimit(1000))
+            .toList();
+
+    List<TestPeer> nonCustodyPeers =
+        craftNodeIdsNotCustodyOf(columnId)
+            .limit(1)
+            .map(nodeId -> new TestPeer(stubAsyncRunner, nodeId, Duration.ofMillis(100)))
+            .peek(testPeer -> testPeer.setCurrentRequestLimit(1000))
+            .toList();
+
+    List<Blob> blobs = Stream.generate(dataStructureUtil::randomBlob).limit(1).toList();
+    BeaconBlock block = blockResolver.addBlock(10, 1);
+    List<DataColumnSidecar> sidecars =
+        miscHelpers.constructDataColumnSidecars(createSigned(block), blobs, kzg);
+    DataColumnSidecar sidecar0 = sidecars.get(columnId.intValue());
+
+    ColumnSlotAndIdentifier id0 = createId(block, columnId.intValue());
+
+    testPeerManager.connectPeer(custodyPeers.get(0));
+    testPeerManager.connectPeer(nonCustodyPeers.get(0));
+
+    SafeFuture<DataColumnSidecar> resp0 = simpleSidecarRetriever.retrieve(id0);
+
+    advanceTimeGradually(Duration.ofSeconds(2));
+
+    assertThat(resp0).isNotDone();
+    assertThat(nonCustodyPeers.get(0).getRequests()).isEmpty();
+    assertThat(custodyPeers.get(0).getRequests()).isNotEmpty();
+
+    custodyPeers.get(1).addSidecar(sidecar0);
+    testPeerManager.connectPeer(custodyPeers.get(1));
+
+    advanceTimeGradually(Duration.ofSeconds(30));
+
+    assertThat(nonCustodyPeers.get(0).getRequests()).isEmpty();
+    // TOFIX: The Retriever continues to request from the custodyPeers.get(0) even while it misses
+    // the column
+    // and the new connected peer custodyPeers.get(1) has it
+    assertThat(resp0).isCompletedWithValue(sidecar0);
```

Comment:
I'd add also checks:
- assertThat(custodyPeers.get(1).getRequests()).isNotEmpty();
- assertThat(custodyPeers.get(0).getRequests().size()).isLessThan(3); // or some other small number to ensure we do try (1) earlier rather than later